pkg/performanceprofile/cmd/render/render.go

```diff
@@ -191,45 +194,110 @@ func render(inputDir, outputDir string) error {
 			return err
 		}
 
-		uid := pp.UID
-		if uid == types.UID("") {
-			uid = uuid.NewUUID()
-		}
-
-		or := []v1.OwnerReference{
-			{
-				Kind:       pp.Kind,
-				Name:       pp.Name,
-				APIVersion: pp.APIVersion,
-				UID:        uid,
-			},
+		if addOwnerRef {
+			err = addOwnerReference(components, pp)
+			if err != nil {
+				return err
+			}
 		}
-
-		for _, componentObj := range components.ToObjects() {
-			componentObj.SetOwnerReferences(or)
+		if len(weakRef) > 0 {
```

Comment:
Hmm, I wonder if this should be `else if`. The weak ref makes not much sense when the real reference is set.

Reply:
Yeah, I am just trying to make the arguments as comfortable as possible (= prevent mistakes). And considering the default (weakref with name) we actually do not have a way to unset it when a full ref is requested.

Reply:
sure, fixed in the latest upload.